jujugui/static/gui/src/app/components/post-deployment/post-deployment.js

```diff
@@ -0,0 +1,250 @@
+    let parsedMetadata = {};
+    let lineByLine;
+    markdown = markdown.trim();
+    // If the first instance of a hr is at position 0 and
+    // there is another hr expect metadata.
+    if (markdown.indexOf('---') === 0 && markdown.indexOf('---', 1) !== -1) {
+      lineByLine = markdown.split('\n');
+      let metadata = [];
+      lineByLine.shift();
+      while(lineByLine.length > 0 && lineByLine[0] !== '---') {
+        metadata.push(lineByLine.shift());
+      }
+      lineByLine.shift();
+
+      metadata.forEach(option => {
+        if (option.indexOf(':') !== -1) {
+          const splitOption = option.split(':');
+          parsedMetadata[splitOption[0].trim()] = splitOption[1].trim();
+        } else {
+          console.error(`${option} does not conform to the metadata format of
+            key: value`);
+        }
+      });
+    }
+
+    return {
+      metadata: parsedMetadata,
+      markdown: lineByLine ? lineByLine.join('\n') : markdown
+    };
+  }
+
+  /**
+    Replace templateTags within the markdown with a pre-defined list.
+
+    @param {String} markdown The raw markdown.
+    @return {String} The replaced string.
+  */
+  replaceTemplateTags(markdown) {
+    let replaced = markdown;
+
+    Object.keys(this.templateTags).forEach(templateTag => {
+      replaced = replaced
+        .split(`{${templateTag}}`)
+        .join(this.templateTags[templateTag].content);
+    });
+
+    return replaced;
+  }
+
+  /**
+    Catches click events on the contents of the modal. This allows replaced
+    templateTags to have React functionality.
+
+    @param {SyntheticEvent} evt The click event.
+  */
+  _handleContentClick(evt) {
+    const templateTag = evt.target.getAttribute('data-templatetag');
+    if (templateTag) {
+      const resolvedVar = this.templateTags[templateTag];
+      if (resolvedVar && resolvedVar.onClick) {
+        resolvedVar.onClick.call(this);
+      }
+    }
+  }
+
+  /**
+    Show the details page of a charm or bundle based on the click.
+  */
+  _handleViewDetails() {
+    this.props.showEntityDetails();
+  }
+
+  _closePostDeployment(evt) {
+    this.props.changeState({
+      postDeploymentPanel: null
+    });
+  }
+
+  render() {
+    let classes = [
+      'post-deployment'
+    ];
+    if (this.state.content) {
+      return (
+        <Panel
+          extraClasses={classes.join(' ')}
+          instanceName="post-deployment"
+          visible={true}>
+          <span className="close" tabIndex="0" role="button"
+            onClick={this._closePostDeployment.bind(this)}>
+            <SvgIcon name="close_16"
+              size="16" />
+          </span>
+          <div onClick={this._handleContentClick.bind(this)}
+            dangerouslySetInnerHTML={{__html: this.state.content}} />
+        </Panel>
+      );
+    } else if (this.state.displayName) {
+      classes.push('post-deployment--simple');
+      return (
+        <Panel
+          extraClasses={classes.join(' ')}
```

Comment:
Could we remove this extra prop and just change the instanceName?

Reply:
see above comment re. bem